kernels/portable/cpu/util/vectorized_math.h

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#pragma once
+
+#include <executorch/runtime/core/exec_aten/util/scalar_type_util.h>
+
+#ifdef ET_USE_PYTORCH_HEADERS
+#include <ATen/cpu/vec/vec.h>
+#endif // ET_USE_PYTORCH_HEADERS
+
+#define _ET_INTERNAL_STD_MATH_FUNC(name) \
+  namespace executorch {                 \
+  inline namespace math {                \
+  using std::name;                       \
+  }                                      \
+  } // namespace executorch
+
+#ifdef ET_USE_PYTORCH_HEADERS
+/**
+ * Internal-usage macro for making a vectorized variant of a unary
+ * function available in the executorch::math namespace.
+ */
+#define ET_INTERNAL_VECTORIZED_FLOAT_UNARY_FUNC(func_name)               \
+  namespace executorch {                                                 \
+  inline namespace math {                                                \
+  template <typename T>                                                  \
+  auto func_name(at::vec::Vectorized<T> vec) {                           \
+    if constexpr (!::executorch::runtime::is_floating_point<T>::value) { \
+      return at::vec::convert<float>(vec).func_name();                   \
```

Comment:
is this a valid thing to do? that is convert say an instance of `at::vec::Vectorized<int8_t>` to float and apply the `func_name`? Maybe i am misunderstanding how this works

Reply:
this should be fixed in the latest update.